addons/addon-base-raas-ui/packages/base-raas-ui/src/models/data-sources/DataSourceStudy.js

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+import _ from 'lodash';
+import { types } from 'mobx-state-tree';
+
+const states = {
+  pending: {
```

Comment:
Is it a good idea define the states of DataSourceStudy and DataSourceAccount once and let the two entity share it?

Reply:
They might look like the same now.  But, I was thinking in terms of managed/unmanaged member accounts in the future.  The account reachability states might differ between studies and accounts.  For example, an account might be partially configured (this could happen when some app roles are working and some are not).  I felt it is save to keep them separate.